taskvine/src/manager/vine_manager.c

```diff
@@ -773,6 +773,50 @@ static void cleanup_worker(struct vine_manager *q, struct vine_worker_info *w)
 	cleanup_worker_files(q, w);
 }
 
+/* Start replicating files that may need replication */
+
+static int recover_temp_files(struct vine_manager *q)
+{
+	char *cached_name = NULL;
+	void *empty_val = NULL;
+	int total_replication_count = 0;
+
+	HASH_TABLE_ITERATE(q->temp_files_to_repilicate, cached_name, empty_val)
+	{
+		struct vine_file *f = hash_table_lookup(q->file_table, cached_name);
+
+		if (f) {
+			int curr_file_replication_cnt = vine_file_replica_table_replicate(q, f);
+
+			if (!curr_file_replication_cnt) {
+				hash_table_remove(q->temp_files_to_repilicate, cached_name);
+			}
+
+			total_replication_count += curr_file_replication_cnt;
+		}
+	}
+
+	return total_replication_count;
+}
+
+/* Insert into hashtable temp files that may need replication. */
+
+static void recover_worker_temp_files(struct vine_manager *q, struct vine_worker_info *w)
```

Comment:
This function doesn't actually do the recovery, so let's call it something like `remember_lost_temp_files` or something along those lines.